javatests/arcs/android/systemhealth/testapp/MemoryStats.kt

```diff
@@ -12,16 +12,34 @@
 package arcs.android.systemhealth.testapp
 
 import android.os.Debug
+import arcs.core.util.performance.MemoryIdentifier
 
-/**
- * Don't use apis of [arcs.core.util.performance.MemoryStats] to track memory stats as
- * the system health test app/services aim to find out memory leakage of Arcs storage stack
- * as well as ProdEx stack where we care about the usage of private-dirty dalvik heap excluding
- * the proportional pss usage counted for boot.img.
- */
+/** Utility of generating statistics of memory footprint. */
 object MemoryStats {
+    /**
+     * App-specific JVM heap usage.
+     *
+     * Don't use [MemoryIdentifier.JAVA_HEAP] as it counts not only app-specific private-dirty
+     * dalvik heap usage but also dirty pages of boot images in current process. The former can
+     * properly represent memory footprint of Arcs-specific JVM stack whereas the latter reflects
+     * how many codes/heaps in boot images are visited and allocated which is not our concern.
+     */
     val appJvmHeapKbytes: Long
         get() = with(Debug.MemoryInfo().apply { Debug.getMemoryInfo(this) }) {
             dalvikPrivateDirty.toLong()
         }
+
+    /**
+     * App-specific native heap usage.
+     */
+    val appNativeHeapKbytes: Long
+        get() = arcs.core.util.performance.MemoryStats.stat(MemoryIdentifier.NATIVE_HEAP)[0]
```

Comment:
optional: you could probably replace this with a static import or an aliased import to void the collision.

Reply:
Done and nice suggestion! I totally forgot alias import ;-)